usr/src/uts/common/fs/zfs/zfs_dir.c

```diff
@@ -175,13 +174,27 @@ zfs_dirent_lock(zfs_dirlock_t **dlpp, znode_t *dzp, char *name, znode_t **zpp,
 	 */
 
 	/*
-	 * Decide if exact matches should be requested when performing
-	 * a zap lookup on file systems supporting case-insensitive
-	 * access.
+	 * When matching we may need to normalize & change case and in addition
+	 * we need to honor the lookup request during the match.  First
+	 * determine if we need to normalize this text by looking for a FIRST
+	 * match, or if this is an EXACT match, bypassing normalization.  Then
```

Comment:
Given that `MT_FIRST` and `MT_EXACT` no longer exist, does `FIRST` and `EXACT` refer to something specific here?